helm/cluster-azure/templates/_kcp.tpl

```diff
@@ -15,6 +15,9 @@ osDisk:
   osType: Linux
 sshPublicKey: {{ include "fake-rsa-ssh-key" $ | b64enc }}
 vmSize: {{ $.Values.controlPlane.instanceType }}
+{{- if ( include "network.subnets.controlPlane.name" $ ) }}
```

Comment:
The condition here is a bit different compared to AzureCluster template

Reply:
it was like you describe at first, i did change it this way so we would not add the `subnetName` field when not needed. 

but i can change it back to have the `include` do the `if condition` and always add the `subnetName` field